ui/app/src/components/secrets/SecretEditorForm.tsx

```diff
@@ -0,0 +1,608 @@
+
+  return (
+    <FormProvider {...form}>
+      <Box
+        sx={{
+          display: 'flex',
+          alignItems: 'center',
+          padding: (theme) => theme.spacing(1, 2),
+          borderBottom: (theme) => `1px solid ${theme.palette.divider}`,
+        }}
+      >
+        <Typography variant="h2">{titleAction} Secret</Typography>
+        <Stack direction="row" spacing={1} sx={{ marginLeft: 'auto' }}>
+          {action === 'read' ? (
+            <>
+              <Button disabled={isReadonly} variant="contained" onClick={() => setAction('update')}>
+                Edit
+              </Button>
+              <Button color="error" disabled={isReadonly} variant="outlined" onClick={onDelete}>
+                Delete
+              </Button>
+              <Divider
+                orientation="vertical"
+                flexItem
+                sx={(theme) => ({
+                  borderColor: theme.palette.grey['500'],
+                  '&.MuiDivider-root': {
+                    marginLeft: 2,
+                    marginRight: 1,
+                  },
+                })}
+              />
+              <Button color="secondary" variant="outlined" onClick={onClose}>
+                Close
+              </Button>
+            </>
+          ) : (
+            <>
+              <Button
+                type="submit"
+                variant="contained"
+                disabled={!form.formState.isValid}
+                onClick={form.handleSubmit(processForm)}
+              >
+                {submitText}
+              </Button>
+              <Button color="secondary" variant="outlined" onClick={handleCancel}>
+                Cancel
+              </Button>
+            </>
+          )}
+        </Stack>
+      </Box>
+      <Stack padding={2} gap={2} sx={{ overflowY: 'scroll' }}>
+        <Stack gap={2} direction="row">
+          <Controller
+            name="metadata.name"
+            render={({ field, fieldState }) => (
+              <TextField
+                {...field}
+                required
+                fullWidth
+                label="Name"
+                InputLabelProps={{ shrink: action === 'read' ? true : undefined }}
+                InputProps={{
+                  disabled: action === 'update',
+                  readOnly: action === 'read',
+                }}
+                error={!!fieldState.error}
+                helperText={fieldState.error?.message}
+                onChange={(event) => {
+                  field.onChange(event);
+                }}
+              />
+            )}
+          />
+        </Stack>
+        <Divider />
+
+        <Box sx={{ width: '100%' }}>
+          <Stack
+            direction="row"
+            alignItems="center"
+            justifyContent="space-between"
+            sx={{ borderBottom: 1, borderColor: 'divider', mb: 2 }}
+          >
+            <MenuTabs aria-label="Secret Authorization Setup" value={tabValue} onChange={handleTabChange}>
```

Comment:
Since the 2 tabs are mutually exclusive, it would be better to change them to `OptionsEditorRadios` instead, as we have for datasources config (proxy vs direct access)

Reply:
RIP my tab stuff, make sense